lib/carto/tree/filterset.js

```diff
@@ -92,6 +92,9 @@ tree.Filterset.prototype.toJS = function(env) {
       val = filter._val.toString(true);
     }
     var attrs = "data";
+    if (op === '=~') {
+      return "(typeof " + attrs + "['" + filter.key.value  + "'] === 'string') && " + attrs + "['" + filter.key.value  + "'].match(" + (val.is === 'string' ? "'" + val.toString().replace(/'/g, "\\'") + "'" : val) + ")";
```

Comment:
Do we need to check the type here? I guess that should be validated in other place.